src/main/java/org/openrewrite/java/migrate/lang/var/UseVarForGenericsConstructors.java

```diff
@@ -86,6 +86,18 @@ public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations v
             return transformToVar(vd, leftTypes, rightTypes);
         }
 
+        private static boolean hasBounds(JavaType type) {
+            if (type instanceof JavaType.Parameterized) {
+                return ((JavaType.Parameterized) type).getTypeParameters().stream()
+                        .map(UseVarForGenericsConstructorsVisitor::hasBounds)
+                        .reduce(false, Boolean::logicalOr);
```

Comment:
In terms of performance we have preference for simpler constructs without Streams, as there's less objects created while the recipe runs, which at scale really starts to matter. Could this be simplified into a for loop with an early return?

Reply:
does this preference also apply to snippets like this?
```java
((J.ParameterizedType) clazz).getTypeParameters().stream()
                                .map(Expression::getType)
                                .filter(Objects::nonNull)
                                .collect(Collectors.toList());
```

Reply:
It depends a bit on where in the flow we find that snippet; I'd commented on this one in particular since it's in the path of determining whether or not to make changes, which will be called repeatedly over time, since we recommend people "continuously migrate", by running idempotent recipes periodically.

If a snippet is only in the path where we are making code changes, then that's assumed to run only once, and we're a little more tolerant to readability over performance. With that background I trust that you can use your own judgement as to which is the case when :)